### ISSUE TITLE
BLIP-2 request:  If it's even possible, can you please provide an official example script of how to get the text(caption) features and image features into the same vector space (e.g. for cross-modal retrieval/search using BLIP-2 models, similar to what we can already do with CLIP.)  Thanks in advance.

### ISSUE THREAD

**Issue by wingz1** [huggingface/transformers#25245]
### System Info

linux, python 3.8+, pytorch '1.13.0+cu116'

### Who can help?

@sgugger

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

N/A

### Expected behavior

N/A

**Comment by wingz1** [comment]
Thanks, I figured that -- I will check the forums!   Indeed those methods do exist in BLIP-2, but those outputs don't share the same dimensionality or mean the same thing as the equivalent commands in CLIP due to the how the model is set up.

**Comment by ydshieh** [comment]
Not really an useful answer, but from the following lines in the modeling file, you can go `language_projection` to get the same dimension. But it's super questionable regarding if this is `the same space` with the meaningful text/image features.

(and yes, further question on this topic should be on the forum)

> self.language_projection = nn.Linear(config.qformer_config.hidden_size, config.text_config.hidden_size)

> ilanguage_model_inputs = self.language_projection(query_output)

>  inputs_embeds = self.language_model.get_input_embeddings()(input_ids)
>  inputs_embeds = torch.cat([language_model_inputs, inputs_embeds], dim=1)

**Comment by ayushtues** [comment]
I was working on integrating BlipDiffusion into diffusers https://github.com/huggingface/diffusers/pull/4388/, which also needs multimodal features. Made a local copy of Blip2Qformer and was modifying in this PR, but having the change integrated into HF will make it much cleaner

**Comment by amyeroberts** [comment]
Great - let's add it into transformers then :) !

**Comment by ayushtues** [comment]
@youssefadr is picking this up in https://github.com/huggingface/transformers/issues/25300, glad to help him if needed

**Comment by youssefadr** [comment]
@ayushtues Yes, I'll open a PR this week asap!

**Comment by jpizarrom** [comment]
Hi @youssefadr 

I hope it is fine that opened a draft PR #25612 to share the progress I did about multimodal features.

The weights of the original blipt2 itm model where converted to Blip2ForImageTextRetrieval
